### PR TITLE
Cmake

### PR DESCRIPTION
**Pull request by chanchett**

### CODE REVIEW

**Comment by rpmoore** [comment]
It looks like you checked in some exe files and a pdb file as well as some other ds3 specific dlls.  Was this intentional?

**Comment by DenverM80** [comment]
I don't see any changes to test/bulk_get.cpp or test/test.cpp - are you going to be able to resolve the valgrind memory violations in a separate request before your contract is up?

==4155== Invalid read of size 8
==4155==    at 0x4C2F79E: memcpy@@GLIBC_2.14 (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==4155==    by 0x40A51E: getFileIndexForChunk(unsigned long*, ds3_str*, checksum_result*) (bulk_get.cpp:63)
==4155==    by 0x40A8A9: checkChunkResponsePartials(_ds3_client*, unsigned int, ds3_get_available_chunks_response*, checksum_result*, unsigned int) (bulk_get.cpp:112)
==4155==    by 0x40CB2D: partial_get::test_method() (bulk_get.cpp:366)
==4155==    by 0x40C7B3: partial_get_invoker() (bulk_get.cpp:338)
==4155==    by 0x40A0D3: boost::unit_test::ut_detail::unused boost::unit_test::ut_detail::invoker<boost::unit_test::ut_detail::unused>::invoke<void (*)()>(void (*&)()) (callback.hpp:56)
==4155==    by 0x409FB3: boost::unit_test::ut_detail::callback0_impl_t<boost::unit_test::ut_detail::unused, void (*)()>::invoke() (callback.hpp:89)
==4155==    by 0x4E9F1F0: ??? (in /usr/lib/x86_64-linux-gnu/libboost_unit_test_framework.so.1.54.0)
==4155==    by 0x4E7A545: boost::execution_monitor::catch_signals(boost::unit_test::callback0<int> const&) (in /usr/lib/x86_64-linux-gnu/libboost_unit_test_framework.so.1.54.0)
==4155==    by 0x4E7AD82: boost::execution_monitor::execute(boost::unit_test::callback0<int> const&) (in /usr/lib/x86_64-linux-gnu/libboost_unit_test_framework.so.1.54.0)
==4155==    by 0x4E9F2F1: boost::unit_test::unit_test_monitor_t::execute_and_translate(boost::unit_test::test_case const&) (in /usr/lib/x86_64-linux-gnu/libboost_unit_test_framework.so.1.54.0)
==4155==    by 0x4E88F93: boost::unit_test::framework_impl::visit(boost::unit_test::test_case const&) (in /usr/lib/x86_64-linux-gnu/libboost_unit_test_framework.so.1.54.0)
==4155==  Address 0xd921410 is 16 bytes inside a block of size 22 alloc'd
==4155==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==4155==    by 0x5339610: g_malloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4002.0)
==4155==    by 0x5350B97: g_strndup (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4002.0)
==4155==    by 0x50E5661: ds3_str_init_with_size (ds3_string.c:28)
==4155==    by 0x50DF30F: xml_get_string.isra.7 (ds3.c:773)
==4155==    by 0x50DFBCB: xml_get_string_from_attribute (ds3.c:779)
==4155==    by 0x50DFBCB: _parse_bulk_object (ds3.c:1257)
==4155==    by 0x50DFBCB: _parse_bulk_objects.isra.23 (ds3.c:1298)
==4155==    by 0x50DFEA2: _parse_master_object_list (ds3.c:1596)
==4155==    by 0x50E315E: ds3_get_available_chunks (ds3.c:2071)
==4155==    by 0x41E001: ensure_available_chunks(_ds3_client const*, ds3_str*) (test.cpp:153)
==4155==    by 0x40C9D8: partial_get::test_method() (bulk_get.cpp:361)
==4155==    by 0x40C7B3: partial_get_invoker() (bulk_get.cpp:338)
==4155==    by 0x40A0D3: boost::unit_test::ut_detail::unused boost::unit_test::ut_detail::invoker<boost::unit_test::ut_detail::unused>::invoke<void (*)()>(void (*&)()) (callback.hpp:56)

**Comment by chanchett** [comment]
Ryan: No, I didn't mean to do that at all. Thought I had only specified new library files but maybe I missed a blanket add I had done.

Denver: Yeah, this pull request is specifically for the cmake stuff; I haven't done any work yet fixing the valgrind errors, but I only have one final left these week and I have all of next week, so I should be on top of that.

**Comment by rpmoore** [comment]
I'm surprised we don't have those excluded in a .gitignore file

**Comment by chanchett** [comment]
Agreed. I'll look at it and add them if not already existing.

File: test/checksum.cpp [hunk 1]
```diff
@@ -22,9 +22,9 @@ unsigned long get_size_by_fd(int fd) {
 
 // Function which compares checksums of the files passed
 bool compare_hash_extended(char* filename_1, char* filename_2, unsigned long num_bytes_to_check, unsigned long offset_1, unsigned long offset_2) {
-    int file_descript_1;
-    int file_descript_2;
-    unsigned long file_size_1,file_size_2;
+    GMappedFile* file_descript_1;
+    GMappedFile* file_descript_2;
+    GError *error=NULL;
```
Comment (rpmoore) [inline test/checksum.cpp]: spacing

File: test/checksum.cpp [hunk 2]
```diff
@@ -32,29 +32,29 @@ bool compare_hash_extended(char* filename_1, char* filename_2, unsigned long num
...
+    file_descript_2 = g_mapped_file_new (filename_2, FALSE, &error);
+    if(error!=NULL) exit(-1);
 
-    file_size_1 = get_size_by_fd(file_descript_1);
-    file_size_2 = get_size_by_fd(file_descript_2);
-
-    file_buffer_1 = static_cast<char*>(mmap(0, file_size_1, PROT_READ, MAP_SHARED, file_descript_1, 0));
+    file_buffer_1 = g_mapped_file_get_contents(file_descript_1);
+    file_buffer_2 = g_mapped_file_get_contents(file_descript_2);
+    if(num_bytes_to_check==0) {
```
Comment (rpmoore) [inline test/checksum.cpp]: spacing around `=`

File: test/checksum.cpp [hunk 3]
```diff
@@ -32,29 +32,29 @@ bool compare_hash_extended(char* filename_1, char* filename_2, unsigned long num
...
+        num_bytes_to_check=MIN(g_mapped_file_get_length (file_descript_1), g_mapped_file_get_length (file_descript_1));
```
Comment (rpmoore) [inline test/checksum.cpp]: Spacing around `=`

File: test/checksum.cpp [hunk 4]
```diff
@@ -32,29 +32,29 @@ bool compare_hash_extended(char* filename_1, char* filename_2, unsigned long num
...
     printf("File to be checked:\t%s\n", filename_2);
 
-#if 0
-    file_descript_1 = open(filename_1, O_RDONLY);
-    if(file_descript_1 < 0) exit(-1);
-
-    file_descript_2 = open(filename_2, O_RDONLY);
-    if(file_descript_2 < 0) exit(-1);
+    file_descript_1 = g_mapped_file_new (filename_1, FALSE, &error);
+    if(error!=NULL) exit(-1);
```
Comment (rpmoore) [inline test/checksum.cpp]: Print an error before exiting, we won't know what happens if you do this.

**Pull request merged**